src/CLIFramework/Extension/ExtensionBase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Hook\Hookable;
+use CLIFramework\Hook\HookHolder;
+use CLIFramework\Exception\ExtensionException;
+
+abstract class ExtensionBase
+    implements Extension, Hookable
+{
+    private $hooks;
+
+    private function getHooks()
```

Comment:
the $container is actually the event container.  we can maintain the instance of the event container in our ServiceContainer instead of using singleton.